src/components/Recipes/AddToMealplanContainer.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react'
+import { AddRecipeTitleToMealplanToggle } from '../Button'
+import imagePathPlus from '../../images/inverted-plus.svg'
+import { v4 as uuidv4 } from 'uuid'
+
+import styled from 'styled-components'
+import { saveToLocal, loadFromLocal } from '../../utils/localStorage'
+
+export default function AddToMealplanContainer({ title }) {
+  const [active, setActive] = useState(() => {
+    const mealplanList = loadFromLocal('mealplanList') || {
+      title: '',
+      items: [{ id: 'initial', item: '', done: false }],
+    }
+
+    const initialState =
+      mealplanList.items.find((meal) => {
+        return meal.item === title
+      }) !== undefined
+    return initialState
+  })
+
+  const addRecipeToMealplan = (title) => {
+    const newMeal = { id: uuidv4(), item: title, done: false }
+    const mealplanList = loadFromLocal('mealplanList') || {
+      title: '',
+      items: [{ id: 'initial', item: '', done: false }],
+    }
+    const newMealplanList = {
+      ...mealplanList,
+      items:
+        mealplanList.items.length === 0 ||
+        (mealplanList.items.length === 1 && mealplanList.items[0].item === '')
+          ? [newMeal]
+          : [...mealplanList.items, newMeal],
+    }
+    saveToLocal('mealplanList', newMealplanList)
+  }
+
+  const deleteRecipeFromMealplan = (title) => {
+    const mealplanList = loadFromLocal('mealplanList') || {
+      title: '',
+      items: [{ id: 'initial', item: '', done: false }],
+    }
+    const newMealplanList = {
+      ...mealplanList,
+      items: mealplanList.items.filter((meal) => meal.item !== title),
+    }
+    saveToLocal('mealplanList', newMealplanList)
+  }
+
+  return (
+    <Container>
+      <AddRecipeTitleToMealplanToggle
+        data-testid="add-to-mealplan"
```

Comment:
Ahh, daher kommt die Test-Id :) Vielleiht da ähnlich zum Komponentennamen "Add-or-remove" oder sowas :)

Reply:
Wurde auch in der Umbenennungsaktion berücksichtigt. :)